homeassistant/components/tuya/alarm_control_panel.py

```diff
@@ -149,8 +149,24 @@ def alarm_state(self) -> AlarmControlPanelState | None:
         """Return the state of the device."""
         # When the alarm is triggered, only its 'state' is changing. From 'normal' to 'alarm'.
         # The 'mode' doesn't change, and stays as 'arm' or 'home'.
-        if self._master_state is not None:
-            if self.device.status.get(self._master_state.dpcode) == State.ALARM:
+        if (
+            self._master_state is not None
+            and self.device.status.get(self._master_state.dpcode) == State.ALARM
+        ):
+            # Check if it's a battery warning
+            is_battery_warning = False
+            if self._alarm_msg_dpcode is not None:
+                encoded_msg = self.device.status.get(self._alarm_msg_dpcode)
+                if encoded_msg:
+                    try:
+                        if "Sensor Low Battery" in b64decode(encoded_msg).decode(
+                            "utf-16be"
+                        ):
+                            is_battery_warning = True
+                    except (ValueError, UnicodeDecodeError):
+                        pass
+            # Only report as triggered if NOT a battery warning
+            if not is_battery_warning:
                 return AlarmControlPanelState.TRIGGERED
```

Comment:
If we do decide to keep the processing, then I think this chunck could be simplified:
```suggestion
            # Only report as triggered if NOT a battery warning
            if (changed_by := self.changed_by) is None or "Sensor Low Battery" not in changed_by:
                return AlarmControlPanelState.TRIGGERED
```

Reply:
Thanks. This looks nice. I just wasn't sure if 'changed_by' is updated before 'alarm_state'. So we don't get into a race-condition here.

Reply:
From the debug-logs it looks like 'changed_by' is getting update from the Tuya server ~100ms before 'alarm_state'.
If this is always the order of events, we could go with the simplified code.